### ISSUE TITLE
Enable padding css on panel element

### ISSUE THREAD

**Issue by cnrudd** [xh/hoist-react#1971]
I just had to help a client setup padding on panel element in a dialog.

The client's first choice of how to do it was:
`<Panel padding='10px' />`
which didn't work because we are removing the padding prop.
https://github.com/xh/hoist-react/blob/develop/desktop/cmp/panel/Panel.js#L76

Am wondering if after removing it from the out div, we could put it back on the xh-vframe that is in the content container.

That is most often where the user would have wanted the padding applied, and it avoids having to write css like:
```
.xh-panel .xh-vframe {
padding: 10px;
}
```

**Comment by amcclain** [comment]
Yes - this has long been something we've wanted - equiv to the `bodyPadding` config in ExtJS world. Partly held up by the sense that we could do more work all around on Panel and its internals.

Another workaround is to nest an explicit inner v/hframe - such as:

```
panel({
  item: vframe({
    className: 'xh-pad',
    items: [...]
  })
});
```

But I think you are right that it could be worth it to detect any of the `padding` props, pull them off, and then apply to the inner frame instead of discarding. I would be in support of that.

**Comment by amcclain** [comment]
Updated the title of this ticket to what I think might be a good target for improving our Panel story here.

We could push further and rethink Panel more, or we could pull back and deal only with, say, padding, but I'd like to propose...

* New `contentLayout` prop to control layout of internal content component - the wrapper placed directly around `items`.  Could take one of three values:
  * `vertical` - current behavior and new default, renders items top to bottom with `vframe`
  * `horizontal` - renders items left to right with `hframe`
  * `scroll` - uses a `div` - no flex, overflow:auto

* New `contentClassName` and `contentStyle` props - applies any given vals to that same content element.

* Special handling for `padding` to always apply it to the content element.  Padding is such a common prop to reach for, and I'm pretty sure this is _always_ what a developer who does wants to happen.  
  * We could support `contentPadding` as the canonical prop and then redirect `padding` to this, if that felt like it was clearer. But I think it will save people annoyance if we make just `padding` work.
  * Yes, you could also do this with `contentStyle: {padding: '10px'}` if we add that prop as proposed, but again, nice to spec padding without extra wrapping and as a simple int.

* Note that we already have an `xh-panel__content` classname that we apply to an internal vbox in Panel that wraps _all_ of the contents - so header, toolbars, etc. It would be unfortunate to have these `contentXXX` props and then another use of "content" in the CSS class structure that's not the same thing. Unfortunately it seems like changing the existing panel classnames would be a breaking change that could effect a number of apps - not sure if we are up for that, or if we can find a new word for the new stuff, or ....  Clearly something we need to discuss and work out before starting on this.

**Comment by amcclain** [comment]
I'd consider this closed by #4221 and #4225
